datenguide_python/query_builder.py

```diff
@@ -189,22 +301,29 @@ def allRegionsQuery(
         if lau:
             region_args["lau"] = str(lau)
 
-        regions = Field("regions", fields=fields, args=region_args)
+        if default_fields:
+            defaults: List[Union[str, "Field"]] = ["id", "name"]
+            fields = defaults + fields
 
+        regions = Field(
+            "regions",
+            fields=fields,
+            args=region_args,
+            return_type=Query._return_type_regions,
+        )
         # add fields page, itemsperPage and total for QueryExecutioner
         return cls(
             start_field=Field(
                 "allRegions",
                 fields=[regions, "page", "itemsPerPage", "total"],
                 args=args,
+                return_type=Query._return_type_allreg,
+                default_fields=default_fields,
             )
         )
 
-    def add_field(self, field: Union[str, Field]) -> Field:
-        return self.start_field.add_field(field)
-
-    def __eq__(self, other):
-        return self.__dict__ == other.__dict__
+    def add_field(self, field: Union[str, Field], default_fields=True) -> Field:
```

Comment:
I'm not sure this works with the allRegions query, because this will add sub-fields to allRegions instead of regions, wont it?

It might be necessary to track two fields. regions for adding, and allRegions for transforming the query to a graphQL query string.

Reply:
yes, true! Fixed it